core/src/repositories/cc/Repository.cc

```diff
@@ -6,15 +6,24 @@
 #include <string>
 
 Repository::Repository(std::string socket_path)
-        : UnixDomainStreamSocketServer(socket_path) {}
+        : UnixDomainStreamSocketServer(socket_path) {this->identifier_=0;}
 
-int Repository::HandleMessage(char *buffer){
+int Repository::HandleMessage(char *buffer,int client_file_descriptor){
     cout << "Handling message: " << buffer << endl;
     DataMessage msg = DataMessage(buffer);
-    ProcessMessage(msg);
+    ProcessMessage(msg,client_file_descriptor);
     return 0;
 }
 
+int Repository::ReplyToClient(DataMessage& message,int client_file_descriptor){
+	//TODO Spencer, can I just arbitrarily set the length of the message? and send it? - Andrew
+    char msg[255] = "";
+	message.Flatten(msg);
+	cout << "Replying to client at " << client_file_descriptor << " with message: " << msg << endl;
+	WriteToSocket(msg,client_file_descriptor);
+	return 1;
+}
+
 //Check if the key is contained in the watch list for the repository.
 bool Repository::WatchListContainsKey(unsigned int key){
```

Comment:
Unrelated to this PR but `watch_list_` should be stored in a map rather then a vector. That would remove the necessity of this method

Reply:
Put it as a TODO